### PR TITLE
Format all code with black and add CI tests

### LINKED ISSUE #31: Consider using the black autoformatter
Following on from discussion in #29, we might want to use a tool (like [black](https://github.com/psf/black)) that handles source code formatting automatically.

### PR DESCRIPTION
**Pull request by HGSilveri**
Adding `black` code formatting to CI tests.

First commit will test the CI tests and will purposefully fail, next commit will follow with all changes in a big-bang fashion.

Closes #31 .

### CODE REVIEW

**Comment by HGSilveri** [comment]
Will wait for PR #199 to be merged.

**Comment by HGSilveri** [comment]
@bejito Can you please check the configuration setting on the first commit?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +27,13 @@ jobs:
...
       - name: Lint with flake8
         run: flake8
-
+  black:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Pulser
+        uses: actions/checkout@v2
+      - name: Check formatting with black
+        uses: psf/black@stable
```
Comment (bejito) [inline .github/workflows/ci.yml:36]: Does this check the formatting or format the code ?

**Review by bejito — APPROVED** [review]
Minor question on the CI, otherwise LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HGSilveri) [reply .github/workflows/ci.yml:36]: By default, it runs: `black --check --diff`. So yeah, it checks the formatting and shows what black would change.

**Pull request merged**